packages/core/src/plugin/vendure-plugin.ts

```diff
@@ -89,13 +90,15 @@ export interface APIExtensionDefinition {
      * }`;
      * ```
      */
-    schema?: DocumentNode | (() => DocumentNode | undefined);
+    schema?: DocumentNode | ((schema?: GraphQLSchema) => DocumentNode | undefined);
     /**
      * @description
      * An array of resolvers for the schema extensions. Should be defined as [Nestjs GraphQL resolver](https://docs.nestjs.com/graphql/resolvers-map)
      * classes, i.e. using the Nest `\@Resolver()` decorator etc.
+     *
+     * Passes the current schema as an optional argument, allowing the resolver to be based on the existing schema.
      */
-    resolvers?: Array<Type<any>> | (() => Array<Type<any>>);
+    resolvers?: Array<Type<any>> | ((schema?: GraphQLSchema) => Array<Type<any>>);
```

Comment:
I am not quire sure where the schema can be passed here.

Reply:
That's a tough one. The `resolvers()` function is called from [here](https://github.com/vendure-ecommerce/vendure/blob/0441a0338aeb7742012e24594feb67935d09b2fc/packages/core/src/plugin/dynamic-plugin-api.module.ts#L19) and ultimately the call comes from [here](https://github.com/vendure-ecommerce/vendure/blob/477fe93e463c4ff56f26033a9459a7efd05d4eac/packages/core/src/api/api-internal-modules.ts#L208):

```ts
/**
 * The internal module containing the Shop GraphQL API resolvers
 */
@Module({
    imports: [ApiSharedModule, ...createDynamicGraphQlModulesForPlugins('shop')],
    providers: [...shopResolvers, ...entityResolvers],
    exports: [...shopResolvers],
})
export class ShopApiModule {}
```

which means at that point we don't have access to the schema object at all. There might be some way to refactor this by using some kind of [dynamic module](https://docs.nestjs.com/fundamentals/dynamic-modules#dynamic-modules) pattern. This would be quite advanced but I _think_ should be possible.